Software/Python/easygopigo3.py

```diff
@@ -2724,71 +2750,81 @@ def __init__(self, port="SERIAL",gpg=None, sensor_type=0):
         except:
             raise
 
+        self.use_mutex = use_mutex
+
     def read_temperature(self):
-        '''
-        Return values may be a float, or error strings
-        TBD: raise errors instead of returning strings
-        import done internally so it's done on a as needed basis only
-        '''
+        """
+        Return the temperature in Celsius degrees.
+
+        :returns: The temperature in Celsius degrees.
+        :rtype: float
+
+        If the sensor isn't plugged in, a ``"Bad reading, try again"`` message is returned.
+        If there is a runtime error, then a ``"Runtime error"`` message is returned.
+
+        """
 
         from di_sensors import DHT
 
-        try:
-            _grab_read()
-            temp = DHT.dht(self.sensor_type)[0]
-        except Exception as e:
-            raise
-        finally:
-            _release_read()
+        _ifMutexAcquire(self.use_mutex)
+        temp = DHT.dht(self.sensor_type)[0]
+        _ifMutexRelease(self.use_mutex)
 
         if temp == -2:
-            return "Bad reading, trying again"
+            return "Bad reading, try again"
         elif temp == -3:
-            return "Run the program as sudo"
+            return "Runtime error"
         else:
             # print("Temperature = %.02fC"%temp)
             return temp
 
     def read_humidity(self):
-        '''
-        Return values may be a float, or error strings
-        TBD: raise errors instead of returning strins
-        '''
+        """
+        Return the humidity as a percentage.
+
+        :returns: Return the humidity as a percentage number from 0% to 100%.
+        :rtype: float
+
+        If the sensor isn't plugged in, a ``"Bad reading, try again"`` message is returned.
+        If there is a runtime error, then a ``"Runtime error"`` message is returned.
+
+        """
         from di_sensors import DHT
 
-        try:
-            _grab_read()
-            humidity = DHT.dht(self.sensor_type)[1]
-        except Exception as e:
-            raise
-        finally:
-            _release_read()
+        _ifMutexAcquire(self.use_mutex)
```

Comment:
this used to be in a try/except. Even if dht sensor doesn't actually raise an error, I think it's best to keep everything the same.

Reply:
You mean about keeping the `ifMutexWhatever` inside the finally block? If so, I already did that, but it doesn't show up.